lib/http1.c

```diff
@@ -623,18 +623,18 @@ static size_t flatten_headers(char *buf, h2o_req_t *req, const char *connection)
 static void proceed_pull(struct st_h2o_http1_conn_t *conn, size_t nfilled)
 {
     h2o_iovec_t buf = {conn->_ostr_final.pull.buf, nfilled};
-    int is_final;
+    h2o_send_state_t send_state;
 
     if (buf.len < MAX_PULL_BUF_SZ) {
         h2o_iovec_t cbuf = {buf.base + buf.len, MAX_PULL_BUF_SZ - buf.len};
-        is_final = h2o_pull(&conn->req, conn->_ostr_final.pull.cb, &cbuf);
+        send_state = h2o_pull(&conn->req, conn->_ostr_final.pull.cb, &cbuf);
```

Comment:
Please set `req->http1_is_persistent` to zero in case an error state is returned (since it is generally an indication that the generator failed to build a complete response).

This also applies to `finalostream_send`.